plugins/modules/zos_job_query.py

```diff
@@ -276,11 +278,31 @@ def run_module():
 
     module = AnsibleModule(argument_spec=module_args, supports_check_mode=True)
 
+    args_def = dict(
+        job_name=dict(type="job_identifier", required=False),
+        owner=dict(type="str", required=False),
+        job_id=dict(type="job_identifier", required=False),
+    )
+
+    try:
+        parser = better_arg_parser.BetterArgParser(args_def)
+        parsed_args = parser.parse_args(module.params)
+        module.params = parsed_args
+    except ValueError as err:
+        module.fail_json(
+            msg='Parameter verification failed.',
+            stderr=str(err)
+        )
+
     if module.check_mode:
         return result
 
     try:
-        name, id, owner = validate_arguments(module.params)
+        name = module.params.get("job_name")
+        id = module.params.get("job_id")
+        owner = module.params.get("owner")
+        if id and owner:
+            raise RuntimeError("Argument Error:job id can not be co-exist with owner")
```

Comment:
I would reword the error message a little bit:
> Job ID and job owner were both supplied, but only one can be at a time.

Or something similar. By the way, why can't they coexist? If this is a limitation, it should be noted in the module doc too.

Reply:
Thanks for the recommendation Ivan but in this case I would advocate to use `mutually_exclusive` option inside module_args provided in argument_spec in the following line `module = AnsibleModule(argument_spec=module_args, supports_check_mode=True)`

Reply:
My ask is to remove this validation in favor of using `mutually_exclusive`